third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -1517,7 +1517,10 @@ impl IntegerValue {
                 return Err(PartialVMError::new(StatusCode::INTERNAL_TYPE_ERROR).with_message(msg));
             },
         };
-        res.ok_or_else(|| PartialVMError::new(StatusCode::ARITHMETIC_ERROR))
+        res.ok_or_else(|| {
+            PartialVMError::new(StatusCode::ARITHMETIC_ERROR)
+                .with_message("Division by zero error".to_string())
+        })
```

Comment:
So, the only error possible is a division by zero error?  Are we sure?

Reply:
the error should come from the checked div so it would be error on div by zero